### ISSUE TITLE
How can I help?

### ISSUE THREAD

**Issue by jakemauer** [goniszewski/grimoire#54]
I've been watching this project with great interest since you posted about it on reddit and I'd like to help make some of the roadmap features a reality. Are there features you're working on currently that you'd like help with QA on? Are there features you know you're not going to tackle in the immediate that you'd like me or others to start on? Thanks for open sourcing all your time and effort!

**Comment by goniszewski** [comment]
Hello @jakemauer! I am very pleased to hear that you are interested in contributing to the project. It would mean a great deal to have another committed soul on board 🧙

Lately, I have been working on resolving some of the main issues that have been reported so far, e.g.,
* Problems related to state management [issue #40](https://github.com/goniszewski/grimoire/issues/40)
* Issues with API/PB communication [issue #49](https://github.com/goniszewski/grimoire/issues/49)
* Documentation (still a lot to do here) [grimoire.pro/docs](https://grimoire.pro/docs/intro)

And of course, there's a lot going on behind the scenes, like [the integration API](https://grimoire.pro/api) (meant to be used with a planned [official browser extension](https://github.com/users/goniszewski/projects/1/views/1?pane=issue&itemId=46043693)) or [annual technical debt payment](https://github.com/goniszewski/grimoire/pull/58) that could use some code review and testing 😅 

Not to mention, that Grimoire still lacks a proper favicon or a logo (low priority, but still something of a significant value).

I do not have as much time to work on the project as I would like to lately, but I would be happy to help you get started and provide guidance along the way. I am also open to any suggestions you might have regarding the project's future.

If you are interested in some issues mentioned above, please let me know, and I will provide you with more details. If nothing catches your eye, we'll find something more suitable for your skills and interests!

**Comment by jakemauer** [comment]
@goniszewski Great! To start I think I can help with some icon/favicon options and documentation improvement. I run my apps as docker containers via docker compose files in portainer and it took me a bit of spelunking through github issues to realize I needed to manually run migrations in the pocketbase container. So at the very least we can work on that and either tweak things so migrations definitely auto-run or that it's documented. 

I also want to try to create a "good enough" bookmarklet as having that will allow me to start to replace Pinboard with Grimoire.

**Comment by goniszewski** [comment]
@jakemauer yes, if the `pb_migrations` directory is not present in the defined location. When deploying from Portainer it's probably the case: the DB won't be populated, and therefore won't run. It's best to have one Docker Image, but that would mean running two processes at once, which is not a good idea (an antipattern even) 😞 

And yeah, an actual icon, help with documentation and a bookmarklet are great ideas for a starting point! I would like to take some time to expand the current descriptions on [grimoire.pro](https://grimoire.pro) when I have a chance, too — hopefully soon enough.

Given the bookmarklet example you'll provide, we can create some sort of util that will generate personalized bookmarklet code with user's credentials hard-coded (browser extension will support more secure ways to authenticate users).
